File: python/896.Monotonic-Array.py
class Solution:
    def isMonotonic(self, nums: List[int]) -> bool:
        isInc = True
        isDec = True
        prev = nums[0]

        for i in range(1, len(nums)):
            curr = nums[i]

            if curr < prev:
                isInc = False
            if curr > prev:
                isDec = False

            prev = curr

            if not isInc and not isDec:
                return False
        
        return True

File: javascript/896.Monotonic-Array.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var isMonotonic = function (nums) {
  let isInc = true;
  let isDec = true;
  let prev = nums[0];

  for (let i = 1; i < nums.length; i++) {
    const curr = nums[i];

    if (curr < prev) {
      isInc = false;
    }

    if (curr > prev) {
      isDec = false;
    }

    if (!isInc && !isDec) {
      return false;
    }

    prev = curr;
  }

  return true;
};
